h/models/group.py

```diff
@@ -190,7 +190,20 @@ def members(self) -> tuple[User, ...]:
         be registered with SQLAlchemy and the changes wouldn't be saved to the
         DB. So this is a read-only property that returns an immutable tuple.
         """
-        return tuple(membership.user for membership in self.memberships)
+        return self.get_members()
+
+    def get_members(self, role: GroupMembershipRoles | None = None) -> tuple[User, ...]:
+        """Return a tuple of this group's members."""
+        if role:
+            memberships = [
+                membership
+                for membership in self.memberships
+                if role in membership.roles
+            ]
+        else:
+            memberships = self.memberships
+
+        return tuple(membership.user for membership in memberships)
```

Comment:
I considered having three separate `@property`'s `moderators`, `admins` and `owners`. These would be easier to use (just `group.owners` instead of `group.get_members(GroupMembershipRoles.OWNER`). But you'd have to keep adding more `@property`'s if we add more roles, and you can't add a property for the `GroupMembershipRoles.MEMBER` role because there's already a property called `members` (which returns all members regardless of role).